CMakeLists.txt

```diff
@@ -43,52 +43,56 @@ vlk_get_header_version()
 
 project(VULKAN_HEADERS LANGUAGES C VERSION ${VK_VERSION_STRING})
 
-add_library(Vulkan-Headers INTERFACE)
-add_library(Vulkan::Headers ALIAS Vulkan-Headers)
-target_include_directories(Vulkan-Headers INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
-
 if (CMAKE_VERSION VERSION_LESS "3.21")
     # https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html
     string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR} PROJECT_IS_TOP_LEVEL)
 endif()
 
+if (NOT PROJECT_IS_TOP_LEVEL)
+    set_directory_properties(PROPERTIES EXCLUDE_FROM_ALL ON)
+endif()
```

Comment:
I'd rather not use this solution since it's not very idiomatic. I.E. I have never seen this approach before. I like CMake and even I find this hard to read.

> Unfortunately, for some CMake versions and project generators, it doesn’t always act as expected and can even
result in broken builds.

This still creates an `INSTALL` target on visual studio generated projects that consume this project with add_subdirectory. This `INSTALL` target installs nothing which is pretty confusing. IE this doesn't work nicely on all platforms.